test/e2e/utils/recordedit-helpers.js

```diff
@@ -390,6 +472,95 @@ exports.testPresentationAndBasicValidation = function(tableParams, isEditMode) {
                     });
                 });
             }
+            
+            if (jsonCols.length > 0) {
+            describe("JSON fields, ", function () {
+                it("should show textarea input for JSON datatype and then set the value", function() {
+                    var columns = tableParams.columns.filter(function(c) { if ((c.type === "json" ) && !c.isForeignKey) return true; });
+                    columns.forEach(function(c) {
+                    chaisePage.recordEditPage.getJSONtextAreaForAcolumn(c.name, recordIndex).then(function(jsonTxtArea) {
+                        if (jsonTxtArea) {
+                            expect(true).toBeDefined();
+                            JSONDataTypeFields.push(jsonTxtArea);
+
+                            if (c.value != undefined) {
+                                expect(jsonTxtArea.getAttribute('value')).toBe(c._value);
+                            }
+
+                            if (isEditMode && (c.generated || c.immutable)) return;
+
+                            chaisePage.recordEditPage.clearInput(jsonTxtArea);
+                            browser.sleep(10);
+
+                            jsonTxtArea.column = c;
+                            
+                            for (i = 0; i < JSONTestParams.length; i++) {
+                                  jsonTxtArea.clear();
+                                  (function(input, expectedValue) {
+                                      c._value = input;
+                                      jsonTxtArea.sendKeys(input);
+                                      chaisePage.recordEditPage.getJSONInputErrorMessage(jsonTxtArea, 'json').then(function(error) {
```

Comment:
For this you should be sending the keys to the input field and then verifying that the value in the input field matches the one that you would expect. As another test after that, you can verify whether an error was thrown or not. Currently, if an error is thrown it will still pass the expectation.